api/odl.py

```diff
@@ -304,21 +344,33 @@ def checkout(self, patron, pin, licensepool, internal_format):
             raise CannotLoan()
 
         # We have successfully borrowed this book.
-        licensepool.licenses_available -= 1
+        if not hold:
+            licensepool.licenses_available -= 1
+        else:
+            licensepool.licenses_reserved -= 1
+            licensepool.patrons_in_hold_queue -= 1
+            _db.delete(hold)
```

Comment:
This would be a natural place to put a simple check that prevents these numbers from going below zero.